### PR TITLE
API: replace SendLoginLogCommand with LoginHook

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
Replace the `SendLoginLogCommand` sent when a user logs in (successfuly or not) by a `LoginHook`. It's a more generic solution allowing to set multiple services to register to this hook and take actions independently

### CODE REVIEW

File: pkg/models/user_auth.go [hunk 1]
```diff
@@ -36,6 +36,15 @@ type ExternalUserInfo struct {
 	IsDisabled     bool
 }
 
+type LoginInfo struct {
+	AuthModule   string
+	User         *User
+	ExternalUser ExternalUserInfo
+	Query        *LoginUserQuery
```
Comment (sakjur) [inline pkg/models/user_auth.go:43]: Do we need to pass the password here?

Other than the IP address, what additional information do we get from sending the query in the hook?

**Review by xlson — APPROVED** [review]
Looks good!

**Review by sakjur — COMMENTED** [review]
I'm not sure about using the same structure for logging in and the hook. There are advantages, but I'd like to strip out sensitive data from the request before sending it to the hook, could we do that somehow? 🤔

File: pkg/models/user_auth.go (hunk 1 above)
Comment (AgnesToulet) [reply pkg/models/user_auth.go:43]: The IP address can easily be retrieved from the request (and it's sent through the hook) so yes, I guess we only need the username used in the login form. I'll change that!

**Pull request merged**